samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -308,6 +316,7 @@ def on_after_transform_template(self, template):
             while (time() - start_time) < self.TEMPLATE_WAIT_TIMEOUT_SECONDS:
```

Comment:
This resets the `throttle` flag. Is the idea that the 2s sleep will give us enough break from calling that we can safely call SAR again? Might be worth commenting about line 319 to make it clear. This method is pretty hard to parse (lots of nesting).